### DISCUSSION TITLE
Should we call Wait method when we call Result property in Tasks? C# Parallel Programming

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#169] [Q&A] [answered]
Hi,

In chapter 17 Parallel Programming, Page 541, You wrote:

```
    public static void TaskWithResultDemo()
    {
        Task<(int Result, int Remainder)> t1 = new(TaskWithResult, (8, 3));
        t1.Start();
        Console.WriteLine(t1.Result);
        t1.Wait();
        Console.WriteLine($"result from task: {t1.Result.Result} {t1.Result.Remainder}");
    }
```

Should we call the Wait method when we call the Result property? In my opinion, we don't need it.
According to Microsoft learn:

> Accessing the property's get accessor blocks the calling thread until the asynchronous operation is complete; it is equivalent to calling the [Wait](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-7.0) method.

**Comment by christiannagel** [comment, marked as answer]
It's page 451 in the book. The text in the book also mentions that the `Result` property of the `Task<T>` type blocks the calling thread, so a `Wait` is not required. With the first `WriteLine` after starting the task, there's also not a `Wait`, and the result is shown.

What's better than using the `Result` property is to use the `await` keyword. This doesn't block the calling thread.
